with 2 .opam files, tag suggests wrong option "-p"

## Setup

We set up:
- a simple repository with 2 opam packages
- and a dune-project with no `(name <name>)` stanza

~~~
> git clone git@github.com:lehy/dune-release-repro.git
[snip]
> cd dune-release-repro
liba/  libb/  CHANGES  dune-project  dune-workspace  liba.opam  libb.opam
> cat dune-project 
(lang dune 2.7)
~~~

## Problem

Problem: running `dune-release tag`suggests invalid (and not
pertinent) option "-p".

~~~
> dune-release tag
dune-release: [ERROR] cannot determine name automatically: use `-p <name>`
> dune-release tag -p toto
dune-release: unknown option `-p'.
Usage: dune-release tag [OPTION]... [VERSION]
Try `dune-release tag --help' or `dune-release --help' for more information.
~~~

Note: if there is only one .opam file present, dune-release correctly
complains about the missing `(name <name>)` stanza.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -46,6 +46,7 @@
 - Fix a bug causing dune-release to exclude all hidden files and folders (starting with `.`) at the
   repository from the distrib archive (#298, @NathanReb)
 - Better report GitHub API errors, all of the error messages reported by the GitHub API are now checked and reported to the user. (#290, @gpetiot)
+- Fix error message when `dune-release tag` cannot guess the project name (#319, @lehy)
 
 ### Security
 

--- a/lib/pkg.ml
+++ b/lib/pkg.ml
@@ -433,7 +433,7 @@ let infer_name dir =
           | Some n -> Ok n
           | None ->
               Logs.err (fun m ->
-                  m "cannot determine name automatically: use `-p <name>`");
+                  m "cannot determine name automatically: use `--name <name>` or add (name <name>) to dune-project");
               exit 1 ) )
 
 let v ~dry_run ?name ?version ?tag ?(keep_v = false) ?delegate ?build_dir

--- a/lib/pkg.ml
+++ b/lib/pkg.ml
@@ -433,7 +433,9 @@ let infer_name dir =
           | Some n -> Ok n
           | None ->
               Logs.err (fun m ->
-                  m "cannot determine name automatically: use `--name <name>` or add (name <name>) to dune-project");
+                  m
+                    "cannot determine name automatically: use `--name <name>` \
+                     or add (name <name>) to dune-project");
               exit 1 ) )
 
 let v ~dry_run ?name ?version ?tag ?(keep_v = false) ?delegate ?build_dir

--- a/tests/bin/tag-2-packages/dune
+++ b/tests/bin/tag-2-packages/dune
@@ -1,0 +1,3 @@
+(mdx
+ (files run.t)
+ (packages dune-release))

--- a/tests/bin/tag-2-packages/run.t
+++ b/tests/bin/tag-2-packages/run.t
@@ -1,0 +1,55 @@
+Set up a project with two packaged libraries, no name in dune-project.
+
+    $ mkdir liba libb
+    $ cat > CHANGES.md << EOF \
+    > ## 0.42.0\
+    > \
+    > - Some other feature\
+    > \
+    > EOF
+    $ echo "let f x = x" > liba/main.ml
+    $ echo "(library (public_name liba))" > liba/dune
+    $ echo "let f x = x" > libb/main.ml
+    $ echo "(library (public_name libb))" > libb/dune
+    $ touch liba.opam libb.opam
+    $ echo "(lang dune 2.7)" > dune-project
+    $ git init . > /dev/null
+    $ git add liba/* libb*/ CHANGES.md *.opam dune-project
+    $ git commit -m 'Commit.' > /dev/null
+
+Check error message about name in dune-project.
+
+    $ dune-release tag -y
+    dune-release: [ERROR] cannot determine name automatically: use `--name <name>` or add (name <name>) to dune-project
+    [1]
+
+Use --name to set name.
+
+    $ cat > CHANGES.md << EOF \
+    > ## 0.43.0\
+    > \
+    > - Some other feature\
+    > \
+    > EOF
+    $ git add CHANGES.md && git commit -m '0.43' > /dev/null
+
+    $ dune-release tag -y --name toto
+    [-] Extracting tag from first entry in CHANGES.md
+    [-] Using tag "0.43.0"
+    [+] Tagged HEAD with version 0.43.0
+
+Use (name <name>) in dune-project (not committed).
+
+    $ cat > CHANGES.md << EOF \
+    > ## 0.44.0\
+    > \
+    > - Some other feature\
+    > \
+    > EOF
+    $ git add CHANGES.md && git commit -m '0.44' > /dev/null
+    $ echo "(name titi)" >> dune-project
+
+    $ dune-release tag -y
+    [-] Extracting tag from first entry in CHANGES.md
+    [-] Using tag "0.44.0"
+    [+] Tagged HEAD with version 0.44.0

